src/terrain/generation/noise.hpp

```diff
@@ -52,7 +57,7 @@ class Noise {
  * layers. The first layer is between -1, and 1. Subsequent have twice the
  * frequency, and amplitude of persistance times the previous amplitude.
  */
-class NoiseGenerator : public Noise {
+class NoiseGenerator : protected Noise {
```

Comment:
You almost always want to have `public` here. Why are you doing `protected`?

Reply:
I changed the get double function to public for testing. I didn't want the three other noise generators to have the same get double function public. get double should be constexpr so making it public won't break anything. I just didn't want to see it three more times because it was unnecessary.

Reply:
Wut?

Remove `get_double`? It's used in `NoiseGenerator`.

> make it public again.

`get_double` was private and now it is public. This statement doesn't make any sense.

Reply:
No

Make `get_double` private.
And then make `Noise` public

Reply:
get_double is defined in Noise

Reply:
I don't want get_double to be private in noise. You are sudgesting this but also saying that Noise should not be a class.

This is working how I want it to be. Why should I change it?

Reply:
If `Noise` is a class, it 
1. should have pure virtual methods to allow for templating, and thusshould be subclassed publicly.
2.  should provide `get_double` as a protected function.

Currently `Noise` only does (1), meaning its essentially a glorified `namespace Noise` with `get_double` inside. So either:
1. Get rid of `Noise` and make `get_double` a static function.
2. Add some pure virtual methods to `Noise` and subclass publicly.

Reply:
First of all, just because an anti-pattern compiles ok doesn’t mean it’s good. 

Second, to quote you
> I changed the get double function to public for testing. I didn't want the three other noise generators to have the same get double function public. 

Why not just make the function private then if you don’t want it to be public?

Reply:
It is private in in all three other classes. Maybe I want it to be public in Noise.

Reply:
> It is private in in all three other classes. Maybe I want it to be public in Noise.

Then you should rethink your class hierarchy.

> Also it's not really an anti-pattern.

Anything where "the compiler doesn't seem to complain" is an anti-pattern. The compiler also doesn't [seem to complain about dereferencing random pointers](https://godbolt.org/z/qzcG8Y3Mo), but that doesn't mean you should do it.